core/sourcehook/sourcehook.h

```diff
@@ -566,6 +570,737 @@ namespace SourceHook
+		{
+
+		}
+
+
+
+	public:	//	Public Interface
+
+		/**
+		*	@brief Add an instance of this hook to the specified interface
+		*
+		*	@param iface the interface pointer to hook
+		*	@param post true when post-hooking, false when pre-hooking.
+		*	@param handler the handler that will be called in place of the original method
+		*	@param mode the hook mode - Hook_Normal will only hook this instance, while others alter the global virtual table.
+		*/
+		HookInstance Add(Interface* iface, bool post, Delegate handler, ::SourceHook::ISourceHook::AddHookMode mode = ISourceHook::AddHookMode::Hook_Normal)
+		{
+			using namespace ::SourceHook;
+			MemFuncInfo mfi = {true, -1, 0, 0};
+			GetFuncInfo(Method, mfi);
+
+			if (mfi.thisptroffs < 0 || !mfi.isVirtual)
+				return {SH, false};
+
+			CMyDelegateImpl* delegate = new CMyDelegateImpl(handler);
+			int id = (*SH)->AddHook(*PL, mode, iface, mfi.thisptroffs, HookCoreImpl::HookManPubFunc, delegate, post);
+
+			return {SH, id};
+		}
+
+		/**
+		*	@brief Remove an existing hook handler from this interface
+		*
+		*	@param iface the interface to be unhooked
+		*	@param post true if this was a post hook, false otherwise (pre-hook)
+		*	@param handler the handler that will be removed from this hook.
+		*/
+		int Remove(Interface* iface, bool post, Delegate handler)
+		{
+			using namespace ::SourceHook;
+			MemFuncInfo mfi = {true, -1, 0, 0};
+			GetFuncInfo(Method, mfi);
+
+			//	Temporary delegate for .IsEqual() comparison.
+			CMyDelegateImpl temp(handler);
+			return (*SH)->RemoveHook(*PL, iface, mfi.thisptroffs, HookCoreImpl::HookManPubFunc, &temp, post);
+		}
+
+	protected:
+		/**
+		 * @brief Configure the hookmangen for this hook manager
+		 * 
+		 * @param store 
+		 * @param hi 
+		 * @return int Zero on success
+		 */
+		static int HookManPubFunc(bool store, ::SourceHook::IHookManagerInfo *hi)
+		{
+			//	Build the MemberFuncInfo for the hooked method
+			GetFuncInfo<MemberMethod>(static_cast<MemberMethod>(Method), Instance.MFI);
+
+			if ((*SH)->GetIfaceVersion() != SH_IFACE_VERSION || (*SH)->GetImplVersion() < SH_IMPL_VERSION)
+				return 1;
+
+			if (store)
+				Instance.HI = hi;
+
+			if (hi) {
+				//	Build a memberfuncinfo for our hook processor.
+				MemFuncInfo our_mfi = {true, -1, 0, 0};
+				GetFuncInfo(&Parent::Hook, our_mfi);
+
+				void* us = reinterpret_cast<void **>(reinterpret_cast<char *>(&Instance) + our_mfi.vtbloffs)[our_mfi.vtblindex];
+				hi->SetInfo(SH_HOOKMAN_VERSION, Instance.MFI.vtbloffs, Instance.MFI.vtblindex, &Instance.HookHandler.Proto, us);
+			}
+
+			return 0;
+		}
+
+	protected:
+		static Result InvokeDelegates(void* self, Args... args)
+		{
+			return HookHandlerImpl::template HookImplCore<SH, Invoker, Parent>(&Instance, self, args...);
+		}
+
+
+	};
+
+
+	//	You're probably wondering what the hell this does.
+	//	https://stackoverflow.com/questions/11709859/how-to-have-static-data-members-in-a-header-only-library/11711082#11711082
+	//	I hate C++.
```

Comment:
I think this was back when I was trying to get this to work on C++11, before I realized that was plain impossible. I'll look into an `inline static`.

Reply:
Going to keep this as a hack for the time being just because I don't understand the semantic differences between this and `inline static`. Would be more than willing to do either I just dont know what is and isn't UD these days :/